machine/v1/types_powervsprovider.go

```diff
@@ -0,0 +1,184 @@
+	// +kubebuilder:validation:=Required
+	Network PowerVSResource `json:"network"`
+
+	// keyPairName is the name of the KeyPair to use for SSH.
+	// The key pair will be exposed to the instance via the instance metadata service.
+	// On boot, the OS will copy the public keypair into the authorized keys for the core user.
+	// +kubebuilder:validation:=Required
+	KeyPairName string `json:"keyPairName"`
+
+	// systemType is the System type used to host the instance.
+	// systemType determines the number of cores and memory that is available.
+	// Few of the supported SystemTypes are s922,e880,e980.
+	// e880 systemType available only in Dallas Datacenters.
+	// e980 systemType available in Datacenters except Dallas and Washington.
+	// When omitted, this means that the user has no opinion and the platform is left to choose a
+	// reasonable default. The current default is s922 which is generally available.
+	// + This is not an enum because we expect other values to be added later which should be supported implicitly.
+	// +optional
+	SystemType string `json:"systemType,omitempty"`
+
+	// processorType is the VM instance processor type.
+	// It must be set to one of the following values: Dedicated, Capped or Shared.
+	// Dedicated: resources are allocated for a specific client, The hypervisor makes a 1:1 binding of a partition’s processor to a physical processor core.
+	// Shared: Shared among other clients.
+	// Capped: Shared, but resources do not expand beyond those that are requested, the amount of CPU time is Capped to the value specified for the entitlement.
+	// if the processorType is selected as Dedicated, then processors value cannot be fractional.
+	// When omitted, this means that the user has no opinion and the platform is left to choose a
+	// reasonable default. The current default is Shared.
+	// +kubebuilder:validation:Enum:="Dedicated";"Shared";"Capped"
+	// +optional
+	ProcessorType string `json:"processorType,omitempty"`
+
+	// processors is the number of virtual processors in a virtual machine.
+	// when the processorType is selected as Dedicated the processors value cannot be fractional.
+	// maximum value for the Processors depends on the selected SystemType.
+	// when SystemType is set to e880 or e980 maximum Processors value is 143.
+	// when SystemType is set to s922 maximum Processors value is 15.
+	// minimum value for Processors depends on the selected ProcessorType.
+	// when ProcessorType is set as Shared or Capped, The minimum processors is 0.5.
+	// when ProcessorType is set as Dedicated, The minimum processors is 1.
+	// When omitted, this means that the user has no opinion and the platform is left to choose a
+	// reasonable default. The default is set based on the selected ProcessorType.
+	// when ProcessorType selected as Dedicated, the default is set to 1.
+	// when ProcessorType selected as Shared or Capped, the default is set to 0.5.
+	// +optional
+	Processors intstr.IntOrString `json:"processors,omitempty"`
```

Comment:
the examples are all numbers, why intorstring?

Reply:
The default is 0.5 which is not an int 😉 This value supports decimal values, but we aren't allowed float types within our APIs. AFAIK the only way to represent a decimal in a Kube API is via a string